src/com/goide/psi/impl/GoPsiImplUtil.java

```diff
@@ -816,8 +816,20 @@ public static GoImportSpec addImportSpec(@NotNull GoImportDeclaration declaratio
     return spec;
   }
 
+  /**
+   * Make local package name valid:
+   * https://github.com/golang/go/blob/master/src/cmd/go/pkg.go#L235-235
+   */
   public static String getLocalPackageName(@NotNull String importPath) {
-    return PathUtil.getFileName(importPath);
+    final String illegalChars = "[-!\"#$%&'()*,:;<=>?[\\]^{|}\uFFFD]";
+    StringBuilder name = new StringBuilder(PathUtil.getFileName(importPath));
```

Comment:
Could be optimized: do not create StringBuilder if no replaces are needed. See com.intellij.openapi.util.text.StringUtil#replaceChar(java.lang.String, char, char) implementation